src/main_thread/decrypt/session_events_listener.ts

```diff
@@ -111,23 +111,23 @@ export default function SessionEventsListener(
         () => runGetLicense(message, messageType),
         backoffOptions,
         manualCanceller.signal,
-      )
-        .then((licenseObject) => {
+      ).then(
+        async (licenseObject) => {
           if (manualCanceller.isUsed()) {
-            return Promise.resolve();
+            return;
           }
           if (isNullOrUndefined(licenseObject)) {
             log.info("DRM: No license given, skipping session.update");
           } else {
             try {
-              return updateSessionWithMessage(session, licenseObject);
+              await updateSessionWithMessage(session, licenseObject);
             } catch (err) {
               manualCanceller.cancel();
               callbacks.onError(err);
             }
           }
-        })
-        .catch((err: unknown) => {
```

Comment:
The idea is to now do `licenseFetchingPromise.then(onSucceed, onFailure)` instead of: `licenseFetchingPromise.then(onSucceed).catch(onFailure)`.